ddtrace/opentelemetry/span.go

```diff
@@ -72,22 +87,22 @@ func (s *span) extractTraceData(c *oteltrace.SpanContextConfig) {
 	state, err := oteltrace.ParseTraceState(headers["tracestate"])
 	if err != nil {
 		log.Debug("Couldn't parse tracestate: %v", err)
+		return
 	}
 	c.TraceState = state
 	parent := strings.Trim(headers["traceparent"], " \t-")
 	if len(parent) > 3 {
 		// checking the length to avoid panic when parsing
-		if f, err := strconv.ParseUint(parent[len(parent)-3:], 16, 8); err != nil {
-			c.TraceFlags = oteltrace.TraceFlags(f)
-		} else {
+		if f, err := strconv.ParseUint(parent[len(parent)-2:], 16, 8); err != nil {
```

Comment:
`len(parent)-2` doesn't mean much to me as a reader. Why 2? In general, a best practice I like is to do something like this instead, which is a bit clearer and less error prone:

`parent[len(parent)-len("dd"):]`
(or whatever that 2 represents, I'm actually not sure)